battle-engine.js

```diff
@@ -2950,9 +2950,11 @@ Battle = (function () {
 			ignoreNegativeOffensive = true;
 			ignorePositiveDefensive = true;
 		}
-
-		var ignoreOffensive = !!(move.ignoreOffensive || (ignoreNegativeOffensive && atkBoosts < 0));
-		var ignoreDefensive = !!(move.ignoreDefensive || (ignorePositiveDefensive && defBoosts > 0));
+		// Check for abilities to see if defenses or offenses are negated (Unaware, Mold Breaker, etc.)
+		var targetAbility = (target.ignore['Ability'] === true || target.ignore['Ability'] === 'A') ? {} : target.getAbility();
+		var pokemonAbility = (pokemon.ignore['Ability'] === true || pokemon.ignore['Ability'] === 'A') ? {} : pokemon.getAbility();
```

Comment:
I don't mind the `(target.ignore['Ability'] === true || target.ignore['Ability'] === 'A')` stuff, but `{}` seems like overkill.

Why not just directly do `var targetAbilityIgnoreOffensive = ...`?